### PR TITLE
Show image pull secret form in a modal

### PR DESCRIPTION
**Pull request by debsmita1**
Fixes bug : https://jira.coreos.com/browse/ODC-1473

image-pull-secret

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx [hunk 1]
```diff
@@ -72,6 +76,11 @@ const ImageSearch: React.FC = () => {
       });
   };
 
+  const handleSave = (name: string) => {
+    setNewImageSecret(name);
+    values.searchTerm && handleSearch(values.searchTerm);
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx]: what if `handleSearch` fails or couldn't search

Comment (debsmita1) [reply frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx]: You mean if the user creates a secret with incorrect credentials? 
This is how it shows for incorrect credentials

Screenshot from 2019-08-01 00-59-17

File: frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-import { Link } from 'react-router-dom';
 import { k8sCreate } from '@console/internal/module/k8s';
 import { ImageStreamImportsModel } from '@console/internal/models';
 import { getPorts } from '@console/internal/components/source-to-image';
 import { useFormikContext, FormikValues } from 'formik';
-import { TextInputTypes } from '@patternfly/react-core';
+import { TextInputTypes, Alert, AlertActionCloseButton } from '@patternfly/react-core';
 import { InputSearchField } from '../../formik-fields';
+import { secretModalLauncher } from '../CreateSecretModal';
+import { SecretTypeAbstraction } from '../../../../../../public/components/secrets/create-secret';
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx]: Done

File: frontend/packages/dev-console/src/components/import/git/SourceSecretSelector.tsx [hunk 3]
```diff
@@ -3,7 +3,8 @@ import { useFormikContext, FormikValues, useField } from 'formik';
 import { FormGroup } from '@patternfly/react-core';
 import SourceSecretDropdown from '../../dropdown/SourceSecretDropdown';
 import { getFieldId } from '../../formik-fields/field-utils';
-import { sourceSecretModalLauncher } from './CreateSourceSecretModal';
+import { secretModalLauncher } from '../CreateSecretModal';
+import { SecretTypeAbstraction } from '../../../../../../public/components/secrets/create-secret';
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/import/git/SourceSecretSelector.tsx]: Done

File: frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx [hunk 4]
```diff
@@ -83,12 +92,31 @@ const ImageSearch: React.FC = () => {
         required
       />
       <div className="help-block" id="image-name-help">
-        To deploy an image from a private repository, you must{' '}
-        <Link to={`/k8s/ns/${values.project.name || 'default'}/secrets/~new/image`}>
+        To deploy an image from a private repository, you must
+        <button
+          className="btn btn-link odc-pull-image-secret__button"
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx]: We should be using pf4 controls.

Comment (debsmita1) [reply frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx]: Done

**Pull request merged**